include/ginkgo/core/base/lin_op.hpp

```diff
@@ -779,6 +779,58 @@ class EnableAbsoluteComputation : public AbsoluteComputable {
 };
 
 
+/**
+ * Adds the operation M <- a I + b M for matrix M, identity
+ * operator I and scalars a and b, where M is the calling object.
+ */
+class ScaledIdentityAddable {
+public:
+    /**
+     * Scales this and adds another scalar times the identity to it.
+     *
+     * @param a  Scalar to multiply the identity operator by before adding.
+     * @param b  Scalar to multiply this before adding the scaled identity to
+     *   it.
+     */
+    virtual void add_scaled_identity(const LinOp* const a,
+                                     const LinOp* const b) = 0;
+};
+
+
+/**
+ * Implements common checks for the operation M <- a I + b M for matrix M,
+ * identity operator I and scalars a and b, where M is the calling object.
+ *
+ * @tparam ConcreteType  The concrete type that should have this operation
+ *                       (the CRTP parameter).
+ */
+template <typename ConcreteType>
+class EnableScaledIdentityAddition : public ScaledIdentityAddable {
+public:
+    /**
+     * Scales this and adds another scalar times the identity to it.
+     *
+     * @param a  Scalar to multiply the identity operator by before adding.
+     * @param b  Scalar to multiply this before adding the scaled identity to
+     *   it.
+     */
+    void add_scaled_identity(const LinOp* const a,
+                             const LinOp* const b) override
+    {
+        GKO_ASSERT_IS_SCALAR(a);
+        GKO_ASSERT_IS_SCALAR(b);
+        auto ae = make_temporary_clone(
+            static_cast<ConcreteType*>(this)->get_executor(), a);
+        auto be = make_temporary_clone(
+            static_cast<ConcreteType*>(this)->get_executor(), b);
```

Comment:
does `static_cast<LinOp*>` work for it? If work, you can use normal feature class not mixin

Reply:
That's a good idea! It appears to work. We might prefer `as<LinOp>` to make it give a better error message in case the user does something strange. I don't think that would have a significant performance impact.